reference/test/base/composition.cpp

```diff
@@ -105,4 +118,124 @@ TYPED_TEST(Composition, AppliesLinearCombinationToVector)
+{
+    /*
+        cmp = [ 2 ] * [ 3 2 ] * [ -1  1  2 ] * [  9  4 ] * [ 1 0 ]
+              [ 1 ]             [  5 -3  0 ]   [  6 -2 ]   [ 0 1 ]
+                                               [ -3  2 ]
+    */
+    using Mtx = typename TestFixture::Mtx;
+    auto cmp = gko::Composition<TypeParam>::create(this->operators.begin(),
+                                                   this->operators.end());
+    auto x = gko::initialize<Mtx>({1.0, 2.0}, this->exec);
+    auto res = clone(x);
+
+    cmp->apply(lend(x), lend(res));
+
+    GKO_ASSERT_MTX_NEAR(res, l({238.0, 119.0}), r<TypeParam>::value);
+}
+
+
+TYPED_TEST(Composition, AppliesLongestLinearCombinationToVector)
+{
+    /*
+        cmp = [ 2 ] * [ 3 2 ] * [ -1  1  2 ] * [  9  4 ] * [ 1 0 ]
+              [ 1 ]             [  5 -3  0 ]   [  6 -2 ]   [ 0 1 ]
+                                               [ -3  2 ]
+    */
+    using Mtx = typename TestFixture::Mtx;
+    auto cmp = gko::Composition<TypeParam>::create(this->operators.begin(),
+                                                   this->operators.end());
+    auto alpha = gko::initialize<Mtx>({3.0}, this->exec);
+    auto beta = gko::initialize<Mtx>({-1.0}, this->exec);
+    auto x = gko::initialize<Mtx>({1.0, 2.0}, this->exec);
+    auto res = clone(x);
+
+    cmp->apply(lend(alpha), lend(x), lend(beta), lend(res));
+
+    GKO_ASSERT_MTX_NEAR(res, l({713.0, 355.0}), r<TypeParam>::value);
+}
+
+
+TYPED_TEST(Composition, AppliesLongestToVectorMultipleRhs)
+{
+    /*
+        cmp = [ 2 ] * [ 3 2 ] * [ -1  1  2 ] * [  9  4 ] * [ 1 0 ]
+              [ 1 ]             [  5 -3  0 ]   [  6 -2 ]   [ 0 1 ]
+                                               [ -3  2 ]
+    */
+    using Mtx = typename TestFixture::Mtx;
+    auto cmp = gko::Composition<TypeParam>::create(this->operators.begin(),
+                                                   this->operators.end());
+    auto x = clone(this->identity);
+    auto res = clone(x);
+
+    cmp->apply(lend(x), lend(res));
+
+    GKO_ASSERT_MTX_NEAR(res, l({{54.0, 92.0}, {27.0, 46.0}}),
+                        r<TypeParam>::value);
+}
+
+
+TYPED_TEST(Composition, AppliesLongestLinearCombinationToVectorMultipleRhs)
+{
+    /*
+        cmp = [ 2 ] * [ 3 2 ] * [ -1  1  2 ] * [  9  4 ] * [ 1 0 ]
+              [ 1 ]             [  5 -3  0 ]   [  6 -2 ]   [ 0 1 ]
+                                               [ -3  2 ]
+    */
+    using Mtx = typename TestFixture::Mtx;
+    auto cmp = gko::Composition<TypeParam>::create(this->operators.begin(),
+                                                   this->operators.end());
+    auto alpha = gko::initialize<Mtx>({3.0}, this->exec);
+    auto beta = gko::initialize<Mtx>({-1.0}, this->exec);
+    auto x = clone(this->identity);
+    auto res = clone(x);
+
+    cmp->apply(lend(alpha), lend(x), lend(beta), lend(res));
+
+    GKO_ASSERT_MTX_NEAR(res, l({{161.0, 276.0}, {81.0, 137.0}}),
+                        r<TypeParam>::value);
+}
```

Comment:
Very nice testing and extensive!
However, I am missing the most basic case (a composition having only a single operator), which was not already present for some reason.